The native compiler on OmniOS is gcc. Modify GNU make to suit.
We don't define GCC_IS_NATIVE since this also modifies the native
lex and yacc defitions and makes the POSIX tests fail.

diff -ru make-4.2.1~/default.c make-4.2.1/default.c
--- make-4.2.1~/default.c	2016-02-28 17:48:44.000000000 +0000
+++ make-4.2.1/default.c	2018-05-14 09:00:13.656611463 +0000
@@ -529,9 +529,9 @@
 # endif /* __MSDOS__ */
     "OBJC", "gcc",
 #else
-    "CC", "cc",
+    "CC", "gcc",
     "CXX", "g++",
-    "OBJC", "cc",
+    "OBJC", "gcc",
 #endif
 
     /* This expands to $(CO) $(COFLAGS) $< $@ if $@ does not exist,
